web-api/src/embedding.rs

```diff
@@ -54,45 +58,136 @@ impl Default for Pipeline {
+            .with_token_size(config.token_size)?
             .with_pooler();
         config.validate()?;
         let embedder = config.build()?;
 
         Ok(Self::Pipeline(embedder))
     }
 
-    #[allow(clippy::unused_async)] // required for sagemaker
+    async fn load_client(config: &Sagemaker) -> Result<Self, SetupError> {
+        let mut config_loader = aws_config::from_env();
+
+        if let Some(region) = &config.aws_region {
+            config_loader = config_loader.region(Region::new(region.clone()));
+        }
+        if let Some(profile) = &config.aws_profile {
+            config_loader = config_loader.profile_name(profile.clone());
+        }
+
+        config_loader = config_loader.retry_config(
+            RetryConfig::standard()
+                .with_max_attempts(1 + config.retry_max_attempts.unwrap_or_default()),
+        );
+
+        let sdk_config = config_loader.load().await;
+        let client = Client::new(&sdk_config);
+
+        Ok(Self::Sagemaker {
+            client,
+            embedding_size: config.embedding_size,
+            endpoint: config.endpoint.clone(),
+            target_model: config.target_model.clone(),
+        })
+    }
+
     pub(crate) async fn run(&self, sequence: &str) -> Result<NormalizedEmbedding, InternalError> {
         match self {
             Embedder::Pipeline(embedder) => embedder
                 .run(sequence)
                 .map_err(InternalError::from_std)?
                 .normalize()
                 .map_err(InternalError::from_std),
+            Embedder::Sagemaker {
+                client,
+                endpoint,
+                target_model,
+                ..
+            } => Self::run_sagemaker(client, endpoint, target_model, sequence).await,
         }
     }
 
+    async fn run_sagemaker(
+        client: &Client,
+        endpoint: &str,
+        target_model: &Option<String>,
+        sequence: &str,
+    ) -> Result<NormalizedEmbedding, InternalError> {
+        let input = json!({
+            "inputs": [sequence],
+        });
+        let mut request = client
+            .invoke_endpoint()
+            .endpoint_name(endpoint)
+            .content_type("application/json")
+            .body(Blob::new(input.to_string()));
+
+        if let Some(target_model) = target_model {
+            request = request.target_model(target_model);
+        };
+
+        let res = request.send().await.map_err(|e| {
+            InternalError::from_message(format!("Failed to request sagemaker endpoint. Error: {e}"))
+        })?;
+
+        let body = res.body().ok_or(InternalError::from_message(
+            "Received sagemaker response without body.",
+        ))?;
+
+        serde_json::from_slice::<SagemakerResponse>(body.as_ref())
+            .map_err(|e| {
+                InternalError::from_message(format!(
+                    "Failed to deserialize sagemaker response body. Error: {e}"
+                ))
+            })?
```

Comment:
:art: this error would already state that it is a deserialization error for sagemaker, so we can save some message bloat.
```suggestion
            .map_err(InternalError::from_std)?
```